web/src/components/ShareMemoDialog.tsx

```diff
@@ -75,6 +82,22 @@ const ShareMemoDialog: React.FC<Props> = (props: Props) => {
     }
   };
 
+  const handleMemoVisibilityOptionChanged = async (visibility: Visibility) => {
+    await memoStore
+      .updateMemo(
+        {
+          id: memo.id,
+          visibility: visibility,
+        },
+        ["visibility"],
+      )
+      .then((newMemo: Memo) => {
+        if (newMemo.visibility == visibility) {
+          toast.success(t("common.changed"));
+        }
+      });
```

Comment:
Don't need to use `then` in `async/await` function.

```suggestion
    const updatedMemo = await memoStore
      .updateMemo(
        {
          id: memo.id,
          visibility: visibility,
        },
        ["visibility"],
      )
      .then((newMemo: Memo) => {
        if (newMemo.visibility == visibility) {
          toast.success(t("common.changed"));
        }
      });
```

Reply:
I force-pushed a new version, please check it.